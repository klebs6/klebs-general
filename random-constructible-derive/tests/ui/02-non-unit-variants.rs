extern crate random_constructible;
extern crate random_constructible_derive;

use random_constructible::*;
use random_constructible_derive::RandomConstructible;

#[derive(RandomConstructible, Copy, Clone, Debug, PartialEq, Eq, Hash)]
enum NonUnitEnum {
    VariantA,
    VariantB(u32),
    VariantC,
}

fn main() {}

