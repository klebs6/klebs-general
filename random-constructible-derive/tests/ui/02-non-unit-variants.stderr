error: proc-macro derive panicked
 --> tests/ui/02-non-unit-variants.rs:7:10
  |
7 | #[derive(RandomConstructible, Copy, Clone, Debug, PartialEq, Eq, Hash)]
  |          ^^^^^^^^^^^^^^^^^^^
  |
  = help: message: RandomConstructible can only be derived for enums with unit variants

warning: unused import: `random_constructible::*`
 --> tests/ui/02-non-unit-variants.rs:4:5
  |
4 | use random_constructible::*;
  |     ^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default
